### PR TITLE
Feature/fastapi rate limiter

### PR DESCRIPTION
**Pull request by simasgrilo**
# Tl;dr
Development of a rate limiter based on the requesting IP. It's already prepared for reverse proxy hosting by checking first if a valid X-Forwarded-For is in the request header. 

# Context
This change introduces a rate limiting procedure for our QR code generator app, once as for now, it was designed to be free and used without requiring registration or anything like that. So, it will check for the IP where the request has originated (checking the host of the Request object, and first checking whether the app is behind some kind of Reverse proxy by verifying the existence of header X-Forwarded-For, which holds the original IP in Nginx, for example. It was built by thinking of using with Redis, but any data store compatible with the ABC RateLimiterInterface can be used

This change:
- introduces a full-fledged rate limiter, compatible with any key data store tha implements RateLimiterInterface 
- decouples the configuration of the rate limiter from the routes and services of the app 
- adds unit tests for the features described here

# Test Plan
- Setup a dev environment for the API at hand
- Try and push more requests than the established. in the unit tests, it was 20 requests per second, so push more than 20 requests
- You will get a response with HTTP Status 429 - Too Many Requests.

# Links
N/A

# Checklist 
- [X] Pull request title is succinct with [tiny] if it’s extra small
- [X] Describes the problem
- [X] Describes the solution (screenshots included if UI changes)
- [X] Has a test plan
- [ ] Contains links to any context (Slack, Figma, JIRA ticket, etc.)
- [X] Code is self reviewed for readability, approach, and edge cases
- [X] Lines changed that may require additional explanation are annotated with an explanation
- [X] Change is ideally < 500 lines if possible. < 150 is ideal.

### CODE REVIEW

File: src/app/ratelim/models/rate_limit_config.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+from pydantic import BaseModel
+
+class RateLimitConfig(BaseModel):
+    """ Model class to model a Rate Limiter configuration object
+
+    Args:
+        BaseModel (pydantic.BaseModel): BaseModel from pydantic on which the model is based
+    """
+    data_store: object #RateLimiterInterface
```
[the remark this answers is not in the document]
Comment (simasgrilo) [reply src/app/ratelim/models/rate_limit_config.py:11]: solved
